bin/run.sh

```diff
@@ -0,0 +1,207 @@
+#!/bin/bash -x
+
+VM_NAME=${1}
+USER=${2}
+PASS=${3}
+URI=${4}
+
+[[ "$VM_NAME" ]] || die "No vm name given"
+[[ "$USER" ]] || die "No username given"
+[[ "$PASS" ]] || die "No password given"
+[[ "$URI" ]] || die "No source uri given"
+
+DOMXML=$VM_NAME.xml
+
+# create libvirt auth file
+cat << EOF > libvirt.auth
+[credentials-vmware]
+authname=$USER
+password=$PASS
+
+[auth-esx-default]
+credentials=vmware
+EOF
+
+# get vm domxml
+export LIBVIRT_AUTH_FILE=libvirt.auth
+virsh -c 'vpx://'$USER'@'$URI'?no_verify=1' dumpxml $VM_NAME >> $DOMXML
+rm -f libvirt.auth
+
+# verify domxml for one disk and one nic
+if [[ $(xmllint --xpath "count(//disk[@type='file']/source)" $DOMXML) -gt 1 ]];
+then
+  die "Only one disk per VM is supported ATM"
+fi
+if [[ $(xmllint --xpath "count(//interface[@type='bridge']/source)" $DOMXML) -gt 1 ]];
+then
+  die "Only one interface per VM is supported"
+fi
+
+# get disk location from domxml
+dcPath=$(xmllint --xpath "/domain/*[local-name()='datacenterpath']/text()" $DOMXML)
+diskSource=($(xmllint --xpath "string(//disk/source/@file)" $DOMXML))
+rm -f $DOMXML
+dsName=${diskSource[0]:1:-1}
+fileName=$(echo ${diskSource[1]} | sed 's/\./-flat./g')
+
+host=$(echo $URI | cut -d'/' -f-1)
+
+# build disk uri
+diskUri='https://'$host'/folder/'$fileName'?dcPath='$dcPath'&dsName='$dsName
+
+# authenticate with vmware
+curl --cookie-jar cookies.txt -q --max-redirs '5' --globoff --head --url $diskUri --user $USER':'$PASS --insecure
+cookie=$(tail -n 1 cookies.txt | cut -f 7 | sed 's/"/\\"/g')
+rm -f cookies.txt
+
+# get disk size
+qemu-img info 'json: { "file.cookie": "vmware_soap_session='$cookie'", "file.sslverify": "off", "file.driver": "https", "file.url": "'$diskUri'", "file.timeout": 2000 }' > output.txt
+size=$(grep -Eow "[0-9]+\.[0-9]+[MG]" output.txt)'i'
+rm -f output.txt
+
+# we support only one disk, it needs to be aligned with DNS-1123
+PVCNAME=$(echo $VM_NAME-disk-1 | sed -r 's/[_.]+/-/g')
+
+ENCODED_PASS=$(echo -n $PASS | base64)
+
+# Create job template
+tee template.yaml <<EOY
+apiVersion: v1
+kind: Template
+metadata:
+  name: v2v-job-template
+  annotations:
+    openshift.io/display-name: "KubeVirt v2v VM Import"
```

Comment:
I'd definetly include the name or identifier of the original name in the temaplte name and template description, as this template is specific to that VM.

Reply:
Agreed, we need to keep in mind that imported vm name maybe different from the source vm since kubernetes/openshift has different rules on what characters are allowed in the name.